### PR TITLE
##3737524 - Onboarding

### PR DESCRIPTION
**Pull request by fpersico**

### CODE REVIEW

File: src/components/LinkButton/__tests__/index.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { shallow } from "enzyme";
+
+import LinkButton from "../";
+
+describe("LinkButton", () => {
+  it("should render correctly", () => {
+    const path = "/path";
+    const children = "Children";
+    const rest = {
+      bsStyle: "primary",
```
Comment (GiorgioNatili) [inline src/components/LinkButton/__tests__/index.test.js]: what is `bs`?

File: src/components/LinkButton/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { Button } from "react-bootstrap";
+
+type Props = {
+  to: string,
+  children: React.Node
+};
+
+const LinkButton = ({ to, children, ...rest }: Props) => (
+  <LinkContainer to={to}>
+    <Button {...rest}>{children}</Button>
```
Comment (GiorgioNatili) [inline src/components/LinkButton/index.js]: there is a reason why we use a space before the first variable between curly braces in the declarations and not in the HTML? Just curious... :)

File: src/components/LinkWithModal/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  showModal: boolean
+};
+
+class LinkWithModal extends Component<Props, State> {
+  state = {
+    showModal: false
+  };
+
+  openModal = (event: SyntheticEvent<HTMLElement>) => {
+    event.stopPropagation();
```
Comment (GiorgioNatili) [inline src/components/LinkWithModal/index.js]: Why do we want to stop the propagation and preventing the browser to react to this click?

File: src/components/LinkWithModal/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    event.preventDefault();
+    this.setState({
+      showModal: true
```
Comment (GiorgioNatili) [inline src/components/LinkWithModal/index.js]: It looks to me that we started manipulating the app state. :)
Any thoughts about using flux?

File: src/components/LinkWithModal/index.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      showModal: false
+    });
+  };
+
+  render() {
+    const { text, modalTitle, children } = this.props;
+    const { showModal } = this.state;
+    return (
+      <div>
+        <a href="#" onClick={this.openModal}>
```
Comment (GiorgioNatili) [inline src/components/LinkWithModal/index.js]: It seems to me that you want to have the mouse arrow changing to a pointer here, can we get the same results with a css (aka `style="cursor:progress"`)?

File: src/screens/HomePage/index.js [hunk 6]
```diff
@@ -1,12 +1,56 @@
 import React, { Component } from "react";
 
+import Joyride from "react-joyride";
+
 import EmptyLayout from "../../layouts/EmptyLayout";
+import LinkButton from "../../components/LinkButton";
+
+const steps = [
```
Comment (GiorgioNatili) [inline src/screens/HomePage/index.js]: The steps should be in an external json file or something similar.

File: src/screens/HomePage/index.js [hunk 7]
```diff
@@ -1,12 +1,56 @@
...
+        >
+          Entra con SPID
+        </LinkButton>
+        <LinkButton
+          to="/spid/request"
+          bsStyle="link"
+          block
+          className="spid-request"
+        >
+          Non hai SPID?
```
Comment (GiorgioNatili) [inline src/screens/HomePage/index.js]: What are you thinking to use for localization? Once I used `react-lazy-locales`, but I am sure you have some better ideas.

File: src/screens/SpidLoginPage/index.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
+import React, { Component } from "react";
+
+import Joyride from "react-joyride";
+
+import EmptyLayout from "../../layouts/EmptyLayout";
+import SpidIpdButton from "../../components/SpidIdpButton";
+
+const idps = [
```
Comment (GiorgioNatili) [inline src/screens/SpidLoginPage/index.js]: Can we move the IDPS in an external file?

File: src/screens/SpidLoginPage/index.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  },
+  {
+    id: "IDP3"
+  },
+  {
+    id: "IDP4"
+  }
+];
+
+const steps = [
```
Comment (GiorgioNatili) [inline src/screens/SpidLoginPage/index.js]: Same for the steps, having them defined outside of the code would be better.

File: src/screens/SpidRequestPage/components/SpidRequestForm/index.js [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+      },
+      () => this.validateField(name, value)
+    );
+  };
+
+  handleSubmit = (event: SyntheticEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+  };
+
+  validateField(fieldName: string, fieldValue: string | boolean) {
```
Comment (GiorgioNatili) [inline src/screens/SpidRequestPage/components/SpidRequestForm/index.js]: There's a need to have a union type and to don't have separate validation messages?

File: src/utils/regexes.js [hunk 11]
```diff
@@ -0,0 +1 @@
+export const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (GiorgioNatili) [inline src/utils/regexes.js]: I love the regex in a separate file!!!

File: src/screens/SpidLoginPage/index.js (hunk 8 above)
Comment (francescopersico) [reply src/screens/SpidLoginPage/index.js]: Will move the IPDS in src/data.

File: src/components/LinkButton/__tests__/index.test.js (hunk 1 above)
Comment (francescopersico) [reply src/components/LinkButton/__tests__/index.test.js]: It is relative to the *b*oot*s*trap style in this case for the button. More at https://react-bootstrap.github.io/components.html#buttons-props.

File: src/components/LinkButton/index.js (hunk 2 above)
Comment (francescopersico) [reply src/components/LinkButton/index.js]: The difference is that in the HTML we are using JSX to specify an attribute. It is also the prettier default.

File: src/components/LinkWithModal/index.js (hunk 3 above)
Comment (francescopersico) [reply src/components/LinkWithModal/index.js]: The normal behaviour in Bootstrap is that clicking on the checkbox label the checkbox is toggled. What we are doing here is open the Popover and stop the propagation of the click event so the checkbox remains untouched.

File: src/components/LinkWithModal/index.js (hunk 4 above)
Comment (francescopersico) [reply src/components/LinkWithModal/index.js]: React redux is already integrated in the app and we will store all the things like user profile, messages and so on in the store. The store is created in src/store/index.js. I have also integrated the REDUX_DEVTOOL_EXTENSION.

File: src/components/LinkWithModal/index.js (hunk 5 above)
Comment (francescopersico) [reply src/components/LinkWithModal/index.js]: Yes at the moment it was just a "bad trick" to get the Privacy rendered as link. It must be changed also for accessibility.

File: src/screens/HomePage/index.js (hunk 6 above)
Comment (francescopersico) [reply src/screens/HomePage/index.js]: Steps will be fetched and served by the Proxy. But as you suggest meanwhile i'll move all the steps inside a file in src/data.

File: src/screens/HomePage/index.js (hunk 7 above)
Comment (francescopersico) [reply src/screens/HomePage/index.js]: I suggest using react-intl.

File: src/screens/SpidLoginPage/index.js (hunk 9 above)
Comment (francescopersico) [reply src/screens/SpidLoginPage/index.js]: Also in src/data

**Comment by GiorgioNatili** [comment]
Let me know by when you can do the small changes we discussed during the standup, I also would love to see codeclimate fixed before merging. :)

File: src/screens/SpidRequestPage/components/SpidRequestForm/index.js (hunk 10 above)
Comment (fpersico) [reply src/screens/SpidRequestPage/components/SpidRequestForm/index.js]: Will split in two function.

**Comment by fpersico** [comment]
The only things left are the react-intl integration and the codeclimate fix. I'll post the updated code today. For react-int let me know if you want it before merging this PR or in a separate/dedicated PR.

**Pull request merged**